targets/targets.json

```diff
@@ -3592,6 +3592,64 @@
         "device_has": ["USTICKER", "LPTICKER", "ANALOGIN", "I2C", "I2C_ASYNCH", "INTERRUPTIN", "PORTIN", "PORTINOUT", "PORTOUT", "PWMOUT", "SERIAL", "SLEEP", "SPI", "SPISLAVE"],
         "device_name": "nRF51822_xxAC"
     },
+    "MCU_NRF52810": {
+        "inherits": ["Target"],
+        "core": "Cortex-M4",
+        "macros": [
+            "BOARD_PCA10040", 
+            "TARGET_NRF52810",
+            "CMSIS_VECTAB_VIRTUAL",
+            "CMSIS_VECTAB_VIRTUAL_HEADER_FILE=\"cmsis_nvic.h\"",
+            "MBED_TICKLESS"
+        ],
+        "device_has": [
+            "ANALOGIN", 
+            "FLASH",
+            "INTERRUPTIN", 
+            "LOWPOWERTIMER", 
+            "PORTIN", 
+            "PORTINOUT", 
+            "PORTOUT", 
+            "PWMOUT", 
+            "RTC", 
+            "SERIAL",
+            "SERIAL_ASYNCH",
+            "SERIAL_FC",
+            "SLEEP", 
+            "STCLK_OFF_DURING_SLEEP",
+            "TRNG"
+        ],        
+        "extra_labels": [
+            "NORDIC",
+            "NRF5x",
+            "NRF52",
+            "SDK_14_2",
+            "SOFTDEVICE_COMMON",
+            "SOFTDEVICE_S112"
+        ],
+        "config": {
+            "lf_clock_src": {
+                "macro_name": "MBED_CONF_NORDIC_NRF_LF_CLOCK_SRC",
+                "help": "Select Low Frequency clock source. Options: NRF_LF_SRC_XTAL, NRF_LF_SRC_SYNTH, and NRF_LF_SRC_RC",
+                "value": "NRF_LF_SRC_XTAL"
+            },
+            "lf_clock_rc_calib_timer_interval": {
+                "macro_name": "MBED_CONF_NORDIC_NRF_LF_CLOCK_CALIB_TIMER_INTERVAL",
+                "value": 16
+            },
+            "lf_clock_rc_calib_mode_config": {
+                "macro_name": "MBED_CONF_NORDIC_NRF_LF_CLOCK_CALIB_MODE_CONFIG",
+                "value": 0
+            }
+        },
+        "OUTPUT_EXT": "hex",
+        "is_disk_virtual": true,
```

Comment:
@andrewleech : planning to add any post build scripts for combining soft device here?

Reply:
I thought that's done already by the mbed build tooling, the softdevice is treated as the bootloader.

Reply:
@marcuschangarm : has this already landed? can you please confirm?

Reply:
The printout during compilation certainly appears so, after linking if reports it's merging the s112 into the region before the app, it already gets the correct flash start address and length (both 0x18000) from the definitions and passes them to the linker.

Reply:
Yes, the bootloader changes are in.